apps/spruce/src/pages/task/taskTabs/TaskHistory/CommitDetailsList/index.tsx

```diff
@@ -33,12 +36,13 @@ const CommitDetailsList: React.FC<CommitDetailsListProps> = ({
                 task={task}
               />
             );
-          } else if (t.inactiveTasks) {
-            // TODO DEVPROD-16174: Replace with Inactive Commits Button.
+          } else if (inactiveTasks && shouldCollapse) {
```

Comment:
`shouldCollapse` is likely redundant (since it's already part of the calculation for `groupTasks`)
```suggestion
          } else if (inactiveTasks) {
```